variants/github_actions_ci/workflows/ci.yml

```diff
@@ -0,0 +1,157 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - production
+concurrency:
+  # Pushing new changes to a branch will cancel any in-progress CI runs
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+env:
+  RAILS_ENV: test
+  DATABASE_URL: postgres://postgres:postgres@localhost:5432/postgres
+  SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
+  SLACK_CHANNEL: TODO
+  SLACK_ICON: https://github.com/ackama.png?size=48
+  SLACK_USERNAME: TODO_APP_NAME CI
+  SLACK_FOOTER: ""
+jobs:
+  js_based_checks:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: '.node-version'
+          cache: "yarn"
+      - run: yarn install --frozen-lockfile
+      - run: yarn run js-lint
+      - run: yarn run format-check
+  ruby_based_checks:
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    services:
+      db:
+        image: postgres:latest
+        ports: ["5432:5432"]
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: postgres
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+      # redis:
+      #   image: redis:latest
+      #   ports:
+      #     - 6379:6379
+      # elasticsearch:
+      #   image: elasticsearch:latest
+      #   env:
+      #     cluster.name: whats-on-test-cluster
+      #     discovery.type: single-node
+      #     ES_JAVA_OPTS: "-Xms1g -Xmx1g"
+      #   ports:
+      #     - 9200:9200
+      #     - 9300:9300
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Install Required OS Packages
+        run: sudo apt-get -yqq install libpq-dev
+      - name: Set up Ruby (version given by .ruby-version)
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+      - run: bundle exec rubocop
+      - run: bundle exec brakeman --run-all-checks --exit-on-warn --format plain .
+      - run: bundle exec bundle audit --update
+      - run: bin/setup
```

Comment:
do we tend to have bin/setup be dev setup? or do we tend to have checks inside the bin/setup script that do 'this is for dev/this is for ci/etc'

Reply:
Good point. `bin/setup` is for dev setup and I don't think we should use it here because it may evolve to include slow steps we don't want in CI e.g. seeding the DB. Will change.